igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java

```diff
@@ -327,14 +339,21 @@ public JobConfiguration getJobConfig(String jobName) {
       return emptyList();
     }
 
-    return client
-        .getBuildService()
+    return client.getBuildService()
         .builds(
             job.getTeamName(),
             job.getPipelineName(),
             job.getName(),
             host.getBuildLookbackLimit(),
-            since);
+            since)
+        .stream()
+        .sorted()
+        .collect(
+            Collectors.toMap(
+                (b) -> b.getNumber(), Function.identity(), (b1, b2) -> b1, LinkedHashMap::new))
```

Comment:
One more thing: Is there a side effect here that I’m overlooking? Looks like this entire collector + the next call to values() and stream() should just be removed?

Reply:
I'm sure there's a prettier way to express this, but I am collapsing the list on the latest build for each major build number. e.g.: `14, 13.1, 13, 12` would collapse into `14, 13.1, 12`

Reply:
(see where I've updated `Build` to be `Comparable` on `decimalNumber`)

Reply:
Gotcha, thanks!